src/pages/shipping.js

```diff
@@ -22,290 +28,733 @@ export const shippingDashboard = (auth, route, goToSpecimenSearch) => {
+    populateBoxesToShipTable(), // 'Select boxes to ship' section
+    addShippingEventListeners();
 
-    for (const box of boxList) {
-      const boxId = box[conceptIds.shippingBoxId];
-      boxIdAndBagsObj[boxId] = box['bags'];
+    hideAnimation();
+}
+
+const addShippingEventListeners = () => {
+    const userName = appState.getState().userName;
+    addEventNavBarShipment("navBarShippingDash", userName);
+    addEventNavBarShippingManifest(userName);
+    addEventBoxSelectListChanged();
+    addEventNavBarBoxManifest("navBarBoxManifest");
+    addEventLocationSelect("selectLocationList", "shipping_location");
+    addEventAddSpecimenToBox();
+    addEventModalAddBox();
+}
+
+const getStoredLocationOnInit = () => {
+    const selectedLocation = JSON.parse(localStorage.getItem('selections'))?.shipping_location ?? 'none';
+    updateShippingStateSelectedLocation(selectedLocation);
+    return selectedLocation;
+}
+
+/**
+ * Populate the 'Available Collections' table.
+ * @param {array} boxList - list of available boxes.
+ * @param {boolean} loadFromState - if true, load data from state instead of fetching from server.
+ * @returns {array} finalizedSpecimenList - the list of specimens with the finalized flag.
+ * @returns {object} availableCollectionsObj - obj to populate the 'Available Collections' table.
+ * If finalizedSpecimenList or availableCollectionsObj are empty (such as initial load), fetched from server.
+ * Note: Orphan panel is currently hidden by request of the product team. Retain for future use.
+ *       Future orphan panel use would require completed state management implementation in the 'currDeleteButton' event listener.
+ */
+const populateAvailableCollectionsList = async (boxList, loadFromState = false) => {
+    let finalizedSpecimenList = appState.getState().finalizedSpecimenList ?? [];
+    let availableCollectionsObj = appState.getState().availableCollectionsObj ?? {};
+
+    if (!loadFromState) {
+        ({ finalizedSpecimenList, availableCollectionsObj } = await getInstituteSpecimensList(boxList));
     }
 
-    response = await getAllBoxes();
-    let allBoxList = response.data;
-    let allBoxIdAndBagsObj = {};
+    const bagIdList = Object.keys(availableCollectionsObj).sort();
 
-    for (const box of allBoxList) {
-        const boxId = box[conceptIds.shippingBoxId];
-        allBoxIdAndBagsObj[boxId] = box['bags']
+    const tableEle = document.getElementById("specimenList");
+    tableEle.innerHTML = buildPopulateSpecimensHeader();
+    
+    let numRows = 1;
+    let orphanBagId;
+
+    for (const bagId of bagIdList) {
+        if (bagId !== "unlabelled") {
+            const rowEle = tableEle.insertRow();
+            rowEle.insertCell(0).innerHTML = bagId;
+            rowEle.insertCell(1).innerHTML = availableCollectionsObj[bagId].length;
+
+            const hiddenChannel = rowEle.insertCell(2)
+            hiddenChannel.innerHTML = JSON.stringify(availableCollectionsObj[bagId]);
```

Comment:
Why `JSON.stringify()` here?
If the content is not html, using textContent is faster than innerHTML.

Reply:
this is existing code, moved from `events.js`, `populateSpecimensList()`
```javascript
for (const bagId of bagIdList) {
        if (bagId != "unlabelled") {
            let rowEle = tableEle.insertRow();
            rowEle.insertCell(0).innerHTML = bagId;
            rowEle.insertCell(1).innerHTML = bagIdAndtubeIdListObj[bagId].length;

            let hiddenChannel = rowEle.insertCell(2)
            hiddenChannel.innerHTML = JSON.stringify(bagIdAndtubeIdListObj[bagId]);
            hiddenChannel.style.display = "none";
            if (numRows % 2 == 0) {
                rowEle.style['background-color'] = "lightgrey";
            }
            numRows += 1;
        } else {
            orphanBagId = bagId;
        }
    }
```